File: 066-加一.cpp
class Solution {
public:
    vector<int> plusOne(vector<int>& digits) {
        //因为只是加1，只有每一位上都是9，才会产生进位而且是10的幂
        for(int i=digits.size()-1;i>=0;i--)
        {
            if(digits[i] < 9)
            {
                digits[i] ++;
                return digits;
            }
            else
            {
                if(i>0)
                    digits[i] = 0;// 9 + 1 = 10
                else//i == 0
                {
                    digits[0] = 1;
                    digits.push_back(0);
                    return digits;
                }
            }
        }
    }
};

File: Java/066-加一.java
class Solution {
    public int[] plusOne(int[] digits) {
        int carry = 1;
        int i = digits.length - 1;
        while (i >= 0 && carry > 0) {
            int sum = digits[i] + carry;
            carry = sum / 10;
            digits[i] = sum % 10;
            --i;
        }
        if (carry > 0) {
            int[] newArray = new int[digits.length + 1];
            System.arraycopy(digits, 0, newArray, 1, digits.length);
            newArray[0] = carry;
            return newArray;
        }
        return digits;
    }
}